recipes/cryptopp/all/conanfile.py

```diff
@@ -65,6 +69,8 @@ def generate(self):
         tc.variables["BUILD_TESTING"] = False
         tc.variables["BUILD_DOCUMENTATION"] = False
         tc.variables["USE_INTERMEDIATE_OBJECTS_TARGET"] = False
+        if self._is_msvc:
+            tc.variables["DISABLE_ASM"] = str(self.settings.arch).startswith("arm")
```

Comment:
I would disable asm if self.settings.arch not in ["x86", "x86_64"], regardless of the compiler.

Reply:
I think doing this would disable all intrinsics except those on x86. See code: https://github.com/weidai11/cryptopp/blob/64ecb2e974bfecd26467c4c301de3bd5ed29cb67/config_asm.h#L237

As I understand, there is a bug in CryptoPP which causes an error when compiling the NEON code with MSVC. Ideally, everything should work without setting `DISABLE_ASM`, as it does when compiling for ARM on other platforms. So, it seems to be a MSVC-only issue.

By the way, I have found out that the CMake file fails to detect when MSVC is used with the ARM architecture. See https://github.com/noloader/cryptopp-cmake/blob/169d9a4c4c573b062e33e4fda51ac10fb2509c24/CMakeLists.txt#L583. If this part is fixed, then the error that I have described above arises.

Reply:
If it does not work for windows ARM, please, update `validate()` method instead. We avoid reaching cmake or any other build system error, if we can preventing when starting the recipe: https://docs.conan.io/en/latest/migrating_to_2.0/recipes.html#the-validate-method

Also, `self._is_msvc` is deprecated. Prefer https://docs.conan.io/en/latest/reference/conanfile/tools/microsoft.html#conan-tools-microsoft-is-msvc

Reply:
@GIGte Please note that I updated the recipe to 8.7.0 in https://github.com/conan-io/conan-center-index/pull/12479. For version 8.7.0, a different CMake wrapper is now used. You may want to check if the problem persists for version 8.7.0 and if yes, open an issue at https://github.com/abdes/cryptopp-cmake.

Reply:
> If it does not work for windows ARM, please, update `validate()` method instead. We avoid reaching cmake or any other build system error, if we can preventing when starting the recipe: [docs.conan.io/en/latest/migrating_to_2.0/recipes.html#the-validate-method](https://docs.conan.io/en/latest/migrating_to_2.0/recipes.html#the-validate-method)

It should work for ARM but fails to compile for some reason. Simply setting `DISABLE_ASM` solves this issue.

vcpkg also sets `DISABLE_ASM` in their port.

Reply:
@chausner I get an error when installing cryptopp/8.7.0 with Visual Studio 2022.

Both commands fail:
`conan install cryptopp/8.7.0@ -s arch=x86_64 -pr:b default -b missing`
`conan install cryptopp/8.7.0@ -s arch=armv8 -pr:b default -b missing`

<details>
<summary>Log</summary>

```
...
C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(460,5): error MSB8020: The build tools for Visual Studio 2019 (Platform Toolset = 'v142') cannot be found. To build using the v142 build tools, please install Visual Studio 2019 build tools.  Alternatively, you may upgrade to the current Visual Studio tools by selecting the Project menu or right-click the solution, and then selecting "Retarget solution".
...
ERROR: cryptopp/8.7.0: Error in build() method, line 126
        cmake.configure(build_script_folder="cryptopp-cmake")
        ConanException: Error 1 while executing cmake -G "Visual Studio 17 2022" -DCMAKE_TOOLCHAIN_FILE="C:/Users/%USER%/.conan/data/cryptopp/8.7.0/_/_/build/5a61a86bb3e07ce4262c80e1510f9c05e9b6d48b/build/generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="C:/Users/%USER%/.conan/data/cryptopp/8.7.0/_/_/package/5a61a86bb3e07ce4262c80e1510f9c05e9b6d48b" -DCRYPTOPP_SOURCES="C:\Users\%USER%\.conan\data\cryptopp\8.7.0\_\_\build\5a61a86bb3e07ce4262c80e1510f9c05e9b6d48b\src" -DCRYPTOPP_BUILD_TESTING="OFF" -DCRYPTOPP_BUILD_DOCUMENTATION="OFF" -DCRYPTOPP_USE_INTERMEDIATE_OBJECTS_TARGET="OFF" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" "C:\Users\%USER%\.conan\data\cryptopp\8.7.0\_\_\build\5a61a86bb3e07ce4262c80e1510f9c05e9b6d48b\src\cryptopp-cmake"
```

</details>

It fails even with build tools v142 installed.

Reply:
Reported the issues: https://github.com/abdes/cryptopp-cmake/issues/23, https://github.com/weidai11/cryptopp/issues/1155

Reply:
Yes, I think it's best to bump the CMake version since you can't build the package otherwise.

By the way, both of the bugs are fixed now: https://github.com/conan-io/conan-center-index/pull/12889#discussion_r972059942

I can just update cryptopp-cmake in this PR and create a patch for version 8.7.0 with this: https://github.com/weidai11/cryptopp/commit/31fa3384160071793bc428a32383938551b3652c

The first change fixes the armv7 build, and the second change additionally fixes the armv8 build. This way, NEON optimizations won't be disabled.